src/App.js

```diff
@@ -1,24 +1,89 @@
-import React, { Component } from 'react';
-import './App.css';
+import React, { Component } from "react";
+import "./App.css";
+import "./grid.css";
+import weather from "./data/weather.json";
+import Search from "./Components/Search";
+import CurrentWeather from "./Components/CurrentWeather";
+import Upcoming from "./Components/Upcoming";
 
 class App extends Component {
-
   constructor(props) {
     super(props);
     this.state = {
+      weather: weather,
+      city: "London",
+      err: ""
     };
   }
+  //json data^
+  componentDidMount() {
+    this.handleClick();
+  }
+
+  changeHandler = event => {
+    this.setState({
+      city: event.target.value
+    });
+  };
+
+  handleClick = () => {
+    console.log("i've been clicked");
+    fetch(
+      `http://api.openweathermap.org/data/2.5/forecast?q=${
+        this.state.city
+      }&cnt=8&units=metric&appid=${"a75d5e4c8cb937be92dab81195a1c637"}`
+    )
+      .then(res => res.json())
+      .then(res => {
+        console.log("anything", res);
+        if (res.cod === "200") {
+          this.setState({
+            weather: res
+          });
+        }
+        if (res.cod !== "200") {
+          this.setState({
+            err: "invalid city name"
+          });
+        }
+      })
+      .catch(err => {
+        if (err) {
+          this.setState({
+            err: "invalid city name"
+          });
+        }
+      });
+  };
+
+  //response: what the user typed.
 
-  render () {
+  render() {
+    // console.log(this.state);
     return (
       <div className="app">
-        <header className="app__header">
-        </header>
-        <main className="app__main">
-        </main>
+        <div className="container">
+          <Search
+            city={this.state.city}
+            changeHandler={this.changeHandler}
+            handleClick={this.handleClick}
+          />
+          {this.state.err ? (
+            <div className="app__main"> {this.state.err} </div>
+          ) : (
+            <main className="app__main">
+              <CurrentWeather weather={this.state.weather} />
+              <Upcoming weather={this.state.weather} />
```

Comment:
A suggestion for refactoring your code:
`Upcoming` needs access to all the weather data inside the `list` because it's looping through it, however does `CurrentWeather` need to received _all_ the weather data? Could you change it so you only pass down the object it is interested in? You'll need to change what you pass down here, and also how `CurrentWeather` handles the data inside CurrentWeather.js

Changing this will demonstrate whether you really understand the passing down of props and the manipulation of JSON data :wink: